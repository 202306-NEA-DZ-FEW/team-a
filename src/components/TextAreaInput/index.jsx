export default function TextAreaInput({
  name,
  placeholder,
  label,
  type,
  handleChange,
  value,
  handleBlur,
  touched,
  error,
  textLength,
  maxLength,
}) {
  return (
    <div className='form-control'>
      <div className='label'>
        <label htmlFor={name} className='label-text'>
          {label}
        </label>
        <label
          className={`label-text-alt ${touched && error ? "text-error" : ""}`}
        >
          {touched && error ? error : ""}
        </label>
      </div>
      <textarea
        className={`textarea no-scrollbar ${
          touched && error ? "textarea-error" : "textarea-primary"
        }  textarea-bordered h-24 w-full bg-base-300`}
        id={name}
        type={type}
        name={name}
        placeholder={placeholder}
        onChange={handleChange}
        value={value}
        onBlur={handleBlur}
      />
      <div className='label'>
        <label htmlFor={name} className='label-text'></label>
        <label
          className={`label-text-alt ${
            textLength >= maxLength ? "text-error" : ""
          }`}
        >
          {textLength}/{maxLength}
        </label>
      </div>
    </div>
  );
}
