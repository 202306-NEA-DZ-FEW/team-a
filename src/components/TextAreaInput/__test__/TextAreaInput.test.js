import renderer from "react-test-renderer";

import TextAreaInput from "..";

it("renders correctly", () => {
  const tree = renderer.create(<TextAreaInput />).toJSON();
  expect(tree).toMatchSnapshot();
});
