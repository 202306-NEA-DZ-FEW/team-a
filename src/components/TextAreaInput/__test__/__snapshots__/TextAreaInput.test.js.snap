// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="form-control"
>
  <div
    className="label"
  >
    <label
      className="label-text"
    />
    <label
      className="label-text-alt "
    />
  </div>
  <textarea
    className="textarea no-scrollbar textarea-primary  textarea-bordered border-opacity-25 h-24 w-full bg-base-200"
  />
  <div
    className="label"
  >
    <label
      className="label-text"
    />
    <label
      className="label-text-alt "
    >
      /
    </label>
  </div>
</div>
`;
